src/openapi/transport/retry.js

```diff
@@ -128,7 +128,11 @@ TransportRetry.prototype.sendTransportCall = function(transportCall) {
                 const isRetryRequest = !(response && response.status) || callOptions.statuses && callOptions.statuses.indexOf(response.status) >= 0;
                 const isWithinRetryLimitOption = callOptions.retryLimit > 0 && transportCall.retryCount < callOptions.retryLimit;
                 const isWithinRetryTimeoutsOption = callOptions.retryTimeouts && transportCall.retryCount < callOptions.retryTimeouts.length;
-                if (isRetryRequest && (isWithinRetryLimitOption || isWithinRetryTimeoutsOption) && !this.isDisposed) {
+
+                if (!response.isNetworkError && isRetryRequest && (

```

Comment:
@lukeapage 

We should only stop the retry in case of network error, if the method type is POST,
else it should be added to a failed request?

Reply:
Can I suggest this:

  *          'delete':{ retryLimit:3 },
  *          'post':{ retryTimeouts: [1000, 1000, 2000, 3000, 5000], statuses: [504] },

becomes

  *          'delete':{ retryLimit:3, retryNetworkError: true },
  *          'post':{ retryTimeouts: [1000, 1000, 2000, 3000, 5000], statuses: [504], retryNetworkError: false },

so we explicitly define in the options whether to retry a network request